src/main/java/games/strategy/engine/data/GameParser.java

```diff
@@ -443,15 +444,17 @@ private Object getInstance(final String className) throws GameParseException {
     Object instance = null;
     try {
       final Class<?> instanceClass = Class.forName(className);
-      instance = instanceClass.newInstance();
+      instance = instanceClass.getDeclaredConstructor().newInstance();
       // a lot can go wrong, the following list is just a subset of potential pitfalls
-    } catch (final ClassNotFoundException cnfe) {
+    } catch (final ClassNotFoundException e) {
       throw new GameParseException(mapName, "Class <" + className + "> could not be found.");
-    } catch (final InstantiationException ie) {
+    } catch (final InstantiationException e) {
       throw new GameParseException(mapName,
-          "Class <" + className + "> could not be instantiated. ->" + ie.getMessage());
-    } catch (final IllegalAccessException iae) {
-      throw new GameParseException(mapName, "Constructor could not be accessed ->" + iae.getMessage());
+          "Class <" + className + "> could not be instantiated. ->" + e.getMessage());
+    } catch (final IllegalAccessException e) {
+      throw new GameParseException(mapName, "Constructor could not be accessed ->" + e.getMessage());
+    } catch (NoSuchMethodException | SecurityException | IllegalArgumentException | InvocationTargetException e) {
```

Comment:
Java 8 previously threw `SecurityException` from `Class#newInstance()`, so I'm not sure it's necessary to start catching it now.   (Do we ever even run with a security manager enabled?)  Has something changed in Java 9 that requires this to be caught when we upgrade?

Not sure if we need to catch `IllegalArgumentException` here because, according to the Javadocs for `Constructor#newInstance()`, it seems it would only occur if there is parameter/argument count mismatch.  Since we requested the zero arg constructor and are invoking it with zero args, doing otherwise would be a programmer error and we should let the code blow up.  But I'm not 100% sure on this reasoning.

In reality, it looks like **all** of the remaining exception types (`ClassNotFoundException`, `InstantiationException`, `IllegalAccessException`, `NoSuchMethodException`, and `InvocationTargetException`) are subclasses of `ReflectiveOperationException`.  I'm not sure there's any benefit to providing separate messages for each type of exception.  It would probably be good enough to have a single message that includes the class name, and then modify `GameParseException` to accept a "cause" argument so the original exception will be captured in the stack trace of the `GameParseException`.  For example:

```java
try {
  final Class<?> instanceClass = Class.forName(className);
  instance = instanceClass.getDeclaredConstructor().newInstance();
} catch (final ReflectiveOperationException e) {
  throw new GameParseException(mapName, "Failed to create instance of class <" + className + ">", e);
}
```